### ISSUE TITLE
Module does not work from nodejs when loaded through yarn

### ISSUE THREAD

**Issue by josephg** [observablehq/plot#430]
When installing plot through yarn, for some reason you end up with d3-array@3.0.1 installed under d3-geo:

```
 $ npm ls d3-array
site2@ /home/seph/src/xxx
└─┬ @observablehq/plot@0.1.0
  └─┬ d3@6.7.0
    ├── d3-array@2.12.1
    ├─┬ d3-contour@2.0.0
    │ └── d3-array@2.12.1 deduped
    ├─┬ d3-geo@2.0.1
    │ └── d3-array@3.0.1 <---- !!!!
    ├─┬ d3-scale@3.3.0
    │ └── d3-array@2.12.1 deduped
    └─┬ d3-time@2.1.1
      └── d3-array@2.12.1 deduped
```

This errors when trying to run the library, because d3-array@3.0.1 is misconfigured for esm:

```javascript
import * as Plot from '@observablehq/plot'
```

```
$ node test.mjs
Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /home/seph/src/blog/site2/node_modules/d3-geo/node_modules/d3-array/src/index.js
require() of ES modules is not supported.
require() of /home/seph/src/blog/site2/node_modules/d3-geo/node_modules/d3-array/src/index.js from /home/seph/src/blog/site2/node_modules/d3-geo/dist/d3-geo.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /home/seph/src/blog/site2/node_modules/d3-geo/node_modules/d3-array/package.json.
```

You get the same error when loading it via commonjs (using require).

You can work around the issue by manually deleting the duplicate d3-array in d3-geo:

```
$ rm -rf node_modules/d3-geo/node_modules/d3-array
```

But it'd be great to sort this out.

**Comment by Fil** [comment]
The bug was introduced by me in  https://github.com/d3/d3-geo/commit/4b0dc76311781e6ea392a407da9214d7e179ae24
~~~js
d3-array node_modules/d3-geo/package.json
    "d3-array": ">=2.5"
~~~

it's been fixed in
https://github.com/d3/d3-geo/commit/54f8890a44826828bd29a81d0bc6dfba9e6786b4
but we didn't publish a new 2.x release.

however, since we're moving to type=module it now says `"d3-array": "2.5.0 - 3"` (https://github.com/d3/d3-geo/commit/1b961c8c9fb663c512e603024a7ba6fed651fc5f)



… it will all stabilize when d3@7 is published.

**Comment by mbostock** [comment]
We should definitely publish a patch fix to d3-geo. I hadn’t realized we hadn’t published the fix in d3/d3-geo@54f8890.

**Comment by mbostock** [comment]
I’ve published d3-geo 2.0.2 to fix the dependency declaration.
